pages/map/index.tsx

```diff
@@ -1,6 +1,148 @@
+    const mapOptions = {
+        mapTypeControl: true,
+        streetViewControl: true
+    }
+
+    return(
+        <div className='shadow' style={{width: "100%", height: "80vh"}}>
+            {center.lat && center.lng ? 
+                <GoogleMapReact
+                    yesIWantToUseGoogleMapApiInternals
+                    bootstrapURLKeys={{key: mapsKey, libraries:[]}}
+                    defaultCenter={defaultCoords}
+                    center={center}
+                    defaultZoom={11}
+                    options={mapOptions}
+                >
+                    <BiStreetView
+                        //@ts-ignore
+                        lat={center.lat}
+                        lng={center.lng}
+                        text="Marker"
+                        size={26}
+                        style={{position:'absolute', transform: 'translate(-50%, -50%)'}}
+                    />
+
+                    {children}
+                </GoogleMapReact>
+                : 
+                <GoogleMapReact
+                    bootstrapURLKeys={{key: mapsKey, libraries:[]}}
+                    defaultCenter={defaultCoords}
+                    defaultZoom={11}
+                    options={mapOptions}
+                >
+                </GoogleMapReact>
+            }
+        </div>
+    )
+}
+
+const MapPageContent = ({children}: MapPageContent) => {
+    const [zip, setZip] = useState('');
+    const [userCoords, setUserCoords] = useState({lat: null, lng: null});
+
+    const changeHandler = (e: any): void => {
+        //Sets zip code state from form value
+        e.preventDefault();
+
+        //@ts-ignore
+        setZip(e.target.value);
+    }
+
+    const getUserCoords = (e: any): void => {
+        //Sets user coordinates as state
+        e.preventDefault();
+
+        if(navigator.geolocation){
+            navigator.geolocation.getCurrentPosition((position) => {
+                let userCoords = {
+                    lat: position.coords.latitude,
+                    lng: position.coords.longitude
+                }
+
+                setUserCoords((prev: any) => {
+                    return {
+                        ...prev,
+                        lat: userCoords.lat,
+                        lng: userCoords.lng
+                    }
+                });
+            });
+        }
+    }
+
+    return(
+        <Container>
+            {children}
+
+            <Row className='mb-2'>
+                <Col>
+                    <Form className='d-flex flex-row justify-content-between' onSubmit={e => e.preventDefault()}>
+                        <Form.Control required type='zip' placeholder='Enter Zip Code...' value={zip} onChange={(e: any) => changeHandler(e)}></Form.Control>
+                        <Button type='submit' className='mx-2'><BiSearchAlt/></Button>
+                        <Button type='submit' onClick={(e: any) => getUserCoords(e)}><BiCurrentLocation/></Button>
```

Comment:
you should be able to simplify this by passing the `onClick` callback directly. for example:

`<Button type="submit" onClick={getUserCoords}>`